[CedricPei] iP

CedricPei opened the pull request:
# DukePlus

**Still in progress**
> Everything is possible

**Features**
1. Add
2. Delete
3. Find
4. Mark & Unmark
5. Undo

**Supported Task Type**
- Todo
- Deadline
- Event

Commit:
Add Gradle support

Commit:
Level 1

Commit:
Level 1

Commit:
Revert "Level 1"

This reverts commit a2b5ffd02b292764de7bb415d6d82d011aa8f8f5.

Commit:
Level 2

Commit:
Level 3

Commit:
Level 4

Commit:
Level 4

Commit:
Level 5
Not using exceptions

Commit:
Level 6
delete

Commit:
refactor<Duke>: Separate Task classes

Commit:
feature: Level-7

Commit:
feature: Level-8

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
refactor: package related classes

Commit:
refactor: extract most methods from Duke

Commit:
refactor: split classes and fix some bugs

Runner classes and components classes function differently.
Fix the bug when entering only one word.

Commit:
Add the Test module to the project

Commit:
Add version1.0 jar to out

Commit:
Some comments attached.

Commit:
Some comments added.

Commit:
More comments added.

Commit:
More comments added.

Commit:
More comments added.

Commit:
Some line space fixed.

Commit:
Some methods extracted to make it more OOP.

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/components/Task.java
#	src/main/java/components/Todo.java
#	src/main/java/runner/Parser.java
#	src/main/java/runner/TaskList.java

Commit:
Refactor: Comments and methods modified to meet Java Standard.

Commit:
feature: Level-9

Commit:
Add some comments.

Jun-How commented on src/main/java/components/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+ * Class for Deadline object.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
```
Perhaps a more descriptive name could be used?
```Suggestion
    protected LocalDate dueDate;
```

Jun-How commented on src/main/java/components/Event.java:
```diff
@@ -0,0 +1,26 @@
+     * @param from The start date of the event.
+     * @param to The end date of the event.
+     */
+    public Event(String msg, String from, String to) {
```
Similar to above, perhaps a more descriptive name could be used, like fromDate and toDate?
Also, I think it would be a good idea to give a complete name "message" instead of "msg" to go with the coding standard.

Jun-How commented on src/main/java/components/Task.java:
```diff
@@ -0,0 +1,82 @@
+ * Class for Task object encapsulate commonalities in subclasses.
+ */
+public class Task {
+    private int done;
```
I think there may introduce the magic number issue here. How about changing the toggle to a boolean type instead?

Jun-How commented on src/main/java/components/Task.java:
```diff
@@ -0,0 +1,82 @@
+    }
+
+    /**
+     * Get the msg of a Task.
```
I think you missed the @return statement in the header, including some in the rest of the codebase.

Jun-How commented on src/main/java/runner/Parser.java:
```diff
@@ -0,0 +1,187 @@
+            String first = segments[0];
+            String arg = segments[1];
+            switch (first) {
+                case "mark":
```
In switch statements, there is no indentation for case clauses by the coding standard, so perhaps alter accordingly?

CedricPei replied:
Yes thanks!

Jun-How reviewed (changes requested):
In terms of coding standards, there are some issues to be resolved.
Some suggestions for improvement given below.
As a positive point, I find that the structure of the code is easy to understand. Keep up!

Shuggan commented on src/main/java/runner/Storage.java:
```diff
@@ -89,4 +80,18 @@ public void saveList() {
+     * @param line Description of a Task.
+     * @param t Task checked.
+     */
+    public void check_add(String line, Task t) {
```
Just a minor stylistic issue. You can use camelCase for "check_add" to something like "addCheckedTask".

Shuggan replied:
The variable names can be more clear and descriptive. For example, "line" and "t" could be changed to "taskDescription" and "task", respectively

Shuggan commented on src/main/java/runner/Ui.java:
```diff
@@ -67,4 +67,24 @@ public static void addMSG(Task t, int n) {
+    /**
+     * Show the TaskList of a Duke.
+     */
+    public static void showList(TaskList t) {
```
In the showList() method, the code to display the task list is duplicated in the findList() method. This can be refactored to a separate method to improve readability and maintainability.

Commit:
Basic GUI achieved.

Commit:
GUI achieved with Duke bot.

Commit:
Add command hello.

Commit:
Refactor the read(info) method and extract findList method.

Commit:
Add Assertions to Existing Code

This commit adds assertions to verify the correctness of the code at key points. The assertions check the validity of intermediate results and function outputs.

The assertions were added after carefully reviewing the existing code and determining the most critical points to ensure proper functioning. The added assertions do not alter the existing code logic or change the outputs, they only serve to validate the existing implementation.

This change should help to prevent future bugs and make the code more maintainable. As the code evolves, these assertions will provide valuable feedback and serve as a safety net for any changes made to the code.

CedricPei committed:
Merge pull request #2 from CedricPei/branch-A-Assertions

Add Assertions to Existing Code

Commit:
Improve Code Quality

This commit makes changes to improve the overall quality of the code. The changes include refactoring, cleaning up code, and adding documentation where needed.

The refactoring efforts were focused on making the code more modular, readable, and maintainable. Unnecessary code was removed and redundant logic was consolidated. Additionally, clear and concise comments were added to explain the logic and purpose of the code, making it easier for others to understand.

These changes should make the code easier to understand, maintain, and debug, while also making it more efficient and reliable. By improving the code quality, we can ensure that it will continue to function correctly and meet the needs of its users over time.

CedricPei committed:
Merge pull request #3 from CedricPei/branch-A-CodeQuality

Improve Code Quality

Commit:
Name of get_list changed to getList.

Commit:
Undo Function Implemented.

Commit:
Command methods extracted to Class Level.

Commit:
More comments added.

Commit:
Name Duke Changed to Riddle

Commit:
Add Ui.png

Commit:
Simple User Guide Accomplished

Commit:
Change absolute path to relative path for release.
